manifests/init.pp

```diff
@@ -72,10 +73,20 @@
       refreshonly => true,
     }
 
-    create_resources('firewalld_port',      $ports)
-    create_resources('firewalld_zone',      $zones)
-    create_resources('firewalld_service',   $services)
-    create_resources('firewalld_rich_rule', $rich_rules)
+    $defaults = { zone => $default_zone }
+
+    if $zones {
+      create_resources('firewalld_zone', $zones)
+    }
+    if $ports {
```

Comment:
True.

But see my reason for adding this above. Further explanation:
As Puppet(3.x) evaluates the type of a variable by the value supplied,  we might end up with boolean here. If I recall correctly,  `create_resources` thoughts a type error in this case.... But I need to check.